src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.plugin.utils.SalesforceClient;
+import io.cdap.plugin.utils.enums.SObjects;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.sql.Date;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+  static Gson gson = new Gson();
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable) throws
+    IOException, InterruptedException {
+    String uniqueRecordId = SalesforceClient.queryObjectId(objectName);
+
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+    List<Object> bigQueryRows = new ArrayList<>();
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    List<JsonObject> sObjectResponse;
+    sObjectResponse = SalesforceClient.queryObject(uniqueRecordId, objectName);
+    return compareSalesforceAndJsonData(sObjectResponse, bigQueryResponse, targetTable);
+  }
+
+  public static boolean validateBQToSalesforceRecordValues(String targetTable, String objectName) throws
```

Comment:
`validateBQToSalesforceRecordValues`, `validateSalesforceToBQRecordValues` - both functions seem to the exact same thing. Is there a reason why we can't merge or move the functionality to a single function?

Reply:
These two functions are used in two different steps here. The first is for SFtoBQ, in which we create a target table in the sink, and the second is for BQtoSF, in which we create a source table using hooks. As a result, the parameters in both functions that we provide are different.

Reply:
I see, the order of input parameters is different to indicate whether we are moving data from bq to salesforce or otherwise. But the functionality is exactly the same inside the function. and this functionality can be moved to a common function and and both `validateBQToSalesforceRecordValues` and `validateSalesforceToBQRecordValues` can call this function internally.